packages/next/src/build/webpack/loaders/next-metadata-route-loader.ts

```diff
@@ -43,20 +44,17 @@ function getStaticRouteCode(resourcePath: string, fileBaseName: string) {
       : process.env.NODE_ENV !== 'production'
       ? cacheHeader.none
       : cacheHeader.longCache
-  return `\
-import fs from 'fs'
-import { fileURLToPath } from 'url'
+  const code = `\
 import { NextResponse } from 'next/server'
 
 const contentType = ${JSON.stringify(getContentType(resourcePath))}
-const resourceUrl = new URL(import.meta.url)
-const filePath = fileURLToPath(resourceUrl).replace(${JSON.stringify(
-    METADATA_RESOURCE_QUERY
-  )}, '')
+const buffer = Buffer.from(${JSON.stringify(
+    fs
+      .readFileSync(resourcePath.replace(METADATA_RESOURCE_QUERY, ''))
```

Comment:
Possible to avoid sync FS operations?

Reply:
switched to async fs in [c1d0d39](https://github.com/vercel/next.js/pull/49323/commits/c1d0d390acc5b7783a11118fdc700fbe2b77220c)